libs/langchain/langchain/llms/huggingface_pipeline.py

```diff
@@ -245,3 +247,62 @@ def _generate(
         return LLMResult(
             generations=[[Generation(text=text)] for text in text_generations]
         )
+
+    async def _astream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[GenerationChunk]:
+        try:
+            from threading import Thread
+
+            from transformers import TextIteratorStreamer
+
+        except ImportError:
+            raise ImportError(
+                "Could not import transformers python package. "
+                "Please install it with `pip install transformers`."
+            )
+
+        # get TextIteratorStreamer from pipeline
+        try:
+            streamer = self.pipeline._forward_params["streamer"]
+
+            if streamer is None:
+                raise ValueError(
+                    "Could not get TextIteratorStreamer from pipeline. "
+                    "Please check your pipeline."
+                )
+            elif type(streamer) is not TextIteratorStreamer:
+                raise ValueError(
+                    "Passed Streamer is not supported. Please use TextIteratorStreamer."
+                    "Please check your pipeline."
+                )
+        except Exception as e:
+            raise ValueError(
+                "Could not get TextIteratorStreamer from pipeline. "
+                "Please check your pipeline."
+            ) from e
+
+        # Prepare the inputs for the model
+        tok = self.pipeline.tokenizer
+        inputs = tok([prompt], return_tensors="pt")
+        inputs = inputs.to('cuda')
+        
+        generation_kwargs = dict(
+            inputs, 
+            **self.pipeline._forward_params
+            )
+        self.pipeline.model.to('cuda')
```

Comment:
I have 4-bit quantization enabled for my pipeline and this threw a value error. 

> ValueError: `.to` is not supported for `4-bit` or `8-bit` bitsandbytes models. Please use the model as it is, since the model has already been set to the correct devices and casted to the correct `dtype`.

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/flameprof.py", line 397, in <module>
    s.runctx(code, globs, None)
  File "/home/user/.pyenv/versions/3.11.7/lib/python3.11/cProfile.py", line 101, in runctx
    exec(cmd, globals, locals)
  File "/home/user/Project/services/chatbot/app/src/chatbot/__main__.py", line 222, in <module>
    asyncio.run(main())
  File "/home/user/.pyenv/versions/3.11.7/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/home/user/.pyenv/versions/3.11.7/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.pyenv/versions/3.11.7/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/user/Project/services/chatbot/app/src/chatbot/__main__.py", line 218, in main
    await asyncio.create_task(generate())
  File "/home/user/Project/services/chatbot/app/src/chatbot/__main__.py", line 213, in generate
    await example_astream()
  File "/home/user/Project/services/chatbot/app/src/chatbot/__main__.py", line 161, in example_astream
    async for message in final_generation_chain.astream(
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 1576, in astream
    async for chunk in self.atransform(input_aiter(), config, **kwargs):
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 1562, in atransform
    async for chunk in self._atransform_stream_with_config(
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 1013, in _atransform_stream_with_config
    async for chunk in iterator:
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 1535, in _atransform
    async for output in final_pipeline:
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 562, in atransform
    async for output in self.astream(final, config, **kwargs):
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/llms/base.py", line 447, in astream
    raise e
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/langchain/llms/base.py", line 436, in astream
    async for chunk in self._astream(
  File "/home/user/Project/services/chatbot/app/src/chatbot/__main__.py", line 65, in _astream
    self.pipeline.model.to("cuda")
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/accelerate/big_modeling.py", line 426, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/accelerate/big_modeling.py", line 426, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/user/.pyenv/versions/project/lib/python3.11/site-packages/transformers/modeling_utils.py", line 2249, in to
    raise ValueError(
ValueError: `.to` is not supported for `4-bit` or `8-bit` bitsandbytes models. Please use the model as it is, since the model has already been set to the correct devices and casted to the correct `dtype`.
```

The following is my config:

```python
bnb_config: BitsAndBytesConfig = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.float16,
    bnb_4bit_use_double_quant=False,
)
```

Just a heads up 👍

Reply:
thanks for the heads up @santosderek!

Reply:
Does it return two outputs? My code snippet is as follows:
```
# all other model configurations are the same
pipe=pipeline(..., streamer=TextStreamer())
llm = HuggingFacePipeline(pipeline = pipe)

# stream the output
result = llm.stream("What are you?")
for token in result:
   print(token)
```

It returns two duplicate sentences as output. Below is an example.

> I am an AI assistant to help you with your tasks. I can answer simple questions for you.
> I am an AI assistant to help you with your tasks. I can answer simple questions for you.

How can I prevent from getting duplicate outputs?